extensions/hooks/_hook_copy_pdbs_to_package.py

```diff
@@ -18,23 +19,32 @@ def post_package(conanfile):
     # Find dumpbin path
     output = StringIO()
     try:
+        program_files = get_env("ProgramFiles(x86)") or get_env("ProgramFiles")
         conanfile.run(
-            r'"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -find "**\dumpbin.exe" -format json',
+            rf'"{program_files}\Microsoft Visual Studio\Installer\vswhere.exe" -find "**\dumpbin.exe" -format json',
             stdout=output, scope="")
+        match = re.search(r'\[(.*?)\]', str(output.getvalue()), re.DOTALL)
```

Comment:
I found some lines in the json oputput that where not json (I believe they where some warnings or something). So I had to find the json list in the output to process it first 🤷‍♂️

Reply:
mmm do you have an example of the output?

Reply:
I do not have it exactly but it was something like:

```
[vcvarsall.bat] Environment initialized for: 'x64'
mkdir: cannot create directory ‘/dev/shm’: Read-only file system

Creating /dev/shm directory failed.
POSIX semaphores and POSIX shared memory will not work

mkdir: cannot create directory ‘/dev/mqueue’: Read-only file system

Creating /dev/mqueue directory failed.
POSIX message queues will not work

[ "this was the actual list of devenv paths",
  "here as well",
  "and here"
]
```
maybe a missconfiguration of my environment?

Reply:
I see, interesting
this is a consequence of of `self.run` running with the bash enabled (so we get all the other output)
mmm
i wonder if we can disable running via the shell (win bash) for a specific call

Reply:
I see no parameter in the ``run`` function itself to override the recipe attribute https://github.com/conan-io/conan/blob/1d8fb4f7289d3676f8165e514d799095a10a3cb2/conans/model/conan_file.py#L340

We could change the attribute just for the hook run calls and revert its value at the end, but I am unsure about that approach and its consequences.